src/app/hooks/useViewTracker/README.md

```diff
@@ -0,0 +1,43 @@
+# useViewTracker hook
+
+The `useViewTracker` hook handles:
+
+- tracking when an element is in view
+- sending the event to ATI
+
+A view/impression event is triggered when:
+
+- 50% of the element is in the viewport for more than 1 second
+- only once per element per page view
+
+The hook returns an object containing a property called `trackRef` that you need to attach to the element you want to track.
+
+### Usage
+
+```jsx
+import useViewTracker from '#hooks/useViewTracker';
+
+const MostRead = ({ pageData }) => {
+  const { trackRef } = useViewTracker({
+    pageData,
+    componentName: 'most-read',
+    campaignName: 'cps_wsoj',
+  });
+
+  return (
+    <div ref={trackRef}>
+      <h2>This is the most read component</h2>
+    </div>
+  );
+};
+```
+
+### Props
+
+| Argument      | Type   | Required | Example                                                                                                          |
+| ------------- | ------ | -------- | ---------------------------------------------------------------------------------------------------------------- |
+| pageData      | object | yes      | The page data used to hydrate the page                                                                           |
+| componentName | string | yes      | The name of the component or an url encoded title of a promo e.g. `most_read` or `This%20is%20a%20promo%20title` |
+| campaignName  | string | yes      | The name of the campaign e.g. `cps_wsoj`                                                                         |
+| format        | string | no       | Can be used to track things like the position of a promo e.g. `[CHD=promo::2]`                                   |
```

Comment:
This format looks like it is interpreted by ati in some way, could we change this param to be more complex e.g.
```
format: { 
  componentType: 'promo'
  position: '2'
}
```
Then internally the hook can take these props and generate our ATI specific string to be interpreted: `[CHD=promo::2]`?

Reply:
Hmm I'm not sure this param will always look like this though

Reply:
Fair enough, looking at it the other way around, maybe we need to make utilities to use when we track our various components so we aren't building up these strings by hand in each component we track? So keep `format` as a string but consider some utilities when we apply this stuff to specific components?

Reply:
@JonBeeb do you know if the format prop will always follow this kind of pattern? e.g. `[CHD=promo::2]` or are there any other variations of it? e.g. `[CHD=link]`

Reply:
@andrewscfc ok we could do that but I think that would have to be in a separate issue, potentially the issues where we are using these hooks in the components.

Reply:
Yeah that makes sense, thanks for checking about this, always better to lock an interface down 👍